eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,93 +3,373 @@
 
-[CmdletBinding(DefaultParameterSetName = 'Default', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
+[CmdletBinding(DefaultParameterSetName = 'Interactive', SupportsShouldProcess = $true, ConfirmImpact = 'Medium')]
 param (
-    [Parameter(Mandatory = $true)]
-    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [string] $ProvisionerApplicationId,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $ProvisionerApplicationId,
 
-    [Parameter(Mandatory = $true)]
-    [string] $ProvisionerApplicationSecret,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string] $ProvisionerApplicationSecret,
 
-    [Parameter(Mandatory = $true)]
-    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [ValidateNotNullOrEmpty()]
-    [string] $TenantId,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $OpensourceApiApplicationId,
 
-    [Parameter(Mandatory = $true)]
-    [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
-    [string] $SubscriptionId,
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $OpensourceApiApplicationTenantId,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [ValidateNotNullOrEmpty()]
+  [string] $OpensourceApiApplicationSecret,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [Parameter(ParameterSetName = 'Interactive')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $TenantId,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [Parameter(ParameterSetName = 'Interactive')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $SubscriptionId,
+
+  [Parameter(ParameterSetName = 'Provisioner')]
+  [string] $GithubAliasCachePath,
+
+  [Parameter()]
+  [ValidateNotNullOrEmpty()]
+  [string] $Environment = "AzureCloud",
+
+  [Parameter()]
+  [switch] $DeleteNonCompliantGroups,
 
-    [Parameter()]
-    [ValidateNotNullOrEmpty()]
-    $Environment = "AzureCloud",
+  [Parameter()]
+  [int] $DeleteAfterHours = 24,
 
-    [Parameter()]
-    [switch] $Force,
+  [Parameter()]
+  [string] $AllowListPath = "$PSScriptRoot/cleanup-allowlist",
 
-    [Parameter(ValueFromRemainingArguments = $true)]
-    $IgnoreUnusedArguments
+  [Parameter()]
+  [switch] $Force,
+
+  [Parameter(ParameterSetName = 'Interactive')]
+  [switch] $Login,
+
+  [Parameter(ValueFromRemainingArguments = $true)]
+  $IgnoreUnusedArguments
 )
 
-&"$PSScriptRoot/../common/scripts/Import-AzModules.ps1"
+Set-StrictMode -Version 3
 
 # Import resource management helpers and override its Log function.
```

Comment:
We should try to just import common.ps1.

Reply:
Looks like common.ps1 doesn't import the Helpers scripts, which I think makes sense as they introduce other dependencies like Az.Keyvault.

Reply:
OK might still be worth importing common.ps1 and keep the resource helpers as an independent import and not add it to common.ps1

Reply:
Yeah we should probably be using the common logger. How about I do a subsequent PR to add common and use the logger?

Reply:
sounds good.